libraries/ms-common/src/watchdog.c

```diff
@@ -0,0 +1,19 @@
+#include "watchdog.h"
+
+void prv_expiry_callback(SoftTimerId timer_id, void *context) {
+  WatchdogStorage *storage = (WatchdogStorage *)context;
+  storage->callback(storage->callback_context);
+}
+
+void watchdog_start(WatchdogStorage *storage, WatchdogTimeout timeout_ms,
+                    WatchdogExpiryCallback callback, void *context) {
+  storage->timeout_ms = timeout_ms;
+  storage->callback = callback;
+  storage->callback_context = context;
+  watchdog_kick(storage);
```

Comment:
If there is another watchdog with `timer_id = 0` it might get kicked, you prolly wanna just start the timer and that's it.